packages/grafana-ui/src/components/uPlot/context.ts

```diff
@@ -51,85 +50,10 @@ export const usePlotPluginContext = (): PlotPluginsContextType => {
   };
 };
 
-// Exposes API for building uPlot config
-
-interface PlotDataAPI {
-  /** Data frame passed to graph, x-axis aligned */
-  data: DataFrame;
-  /** Returns field by index */
-  getField: (idx: number) => Field;
-  /** Returns x-axis fields */
-  getXAxisFields: () => Field[];
-  /** Returns x-axis fields */
-  getYAxisFields: () => Field[];
-  /** Returns field value by field and value index */
-  getFieldValue: (fieldIdx: number, rowIdx: number) => any;
-  /** Returns field config by field index */
-  getFieldConfig: (fieldIdx: number) => FieldConfig;
-}
-
-export const usePlotData = (): PlotDataAPI => {
```

Comment:
No longer this. Plugins should accept the original data frame and inspect it using GraphNGContext which allows x/y fields inspection using matches.